extensions/http-curl/tests/AbsoluteTimeoutTest.cpp

```diff
@@ -47,4 +47,6 @@ int main() {
   plan->runNextProcessor();
 
   assert(LogTestController::getInstance().contains("HTTP operation timed out, with absolute timeout 3000ms"));
+
+  delete server;
```

Comment:
Could you explain why does leaking it cause segfault?

Btw I think unique_ptr would be a nicer solution.

Reply:
it was just a temporary check, in no way did I intend it to be the solution, but the cause is: if the handler returns after a server shutdown was initiated it will try to consume the next socket and since that signal will never be notified by the server (as it is waiting for the handlers to shutdown 🤦) we have a deadlock

Reply:
the relevant part in `civetweb.c:16178`:

```
	/* Call consume_socket() even when ctx->stop_flag > 0, to let it
	 * signal sq_empty condvar to wake up the master waiting in
	 * produce_socket() */
	while (consume_socket(ctx, &conn->client, conn->thread_index)) {
		conn->conn_birth_time = time(NULL);
```

Reply:
but the master is not waiting in the `produce_socket`

Reply:
```
static int
consume_socket(struct mg_context *ctx, struct socket *sp, int thread_index)
{
	DEBUG_TRACE("%s", "going idle");
	ctx->client_socks[thread_index].in_use = 0;
	event_wait(ctx->client_wait_events[thread_index]);
	*sp = ctx->client_socks[thread_index];
	DEBUG_TRACE("grabbed socket %d, going busy", sp ? sp->sock : -1);

	return !ctx->stop_flag;
}
```

Reply:
the segfault was (probably) caused by the `CivetServer` creating a load of threads and we just exited `main` like there is no tomorrow